### DISCUSSION TITLE
File dowload response replaces greek characters from the filename with underscores

### DISCUSSION THREAD

**Discussion by fractalbit** [livewire/livewire#4946] [7. Bugs (archived)]
When you return a file download response from a livewire method it replaces the greek characters from the filename with underscores. This will probably happen with other non-english characters too. The same code from a laravel controller method does not have the same problem.

```
 $filepath = storage_path('/app/exports/Ελληνικά.xlsx');
 return response()->download($filepath);
```

The above code will return a file named '________.xlsx' which downloads just fine, but the filename is not the desired.

Laravel version: 8.83.11
Livewire-version: 2.10.5

**Comment by filunK** [comment]
I faced the same problem. I tried to download a file with Japanese characters, the multibyte character set.

I afraid ``SupportFileDownloads`` class in ``livewire/src/Features/SupportFileDownloads.php`` makes the unexpected behavior.

A method ``getFilenameFromContentDispositionHeader`` does not refer a attribute ``filename*``, and the attribute sets non-ascii filename.
